Mathlib/Geometry/Convex/Cone/Basic.lean

```diff
@@ -434,77 +397,63 @@ variable [Semiring R] [PartialOrder R] [IsOrderedRing R]
-  carrier := S
-  smul_mem' c _ _ hx := S.smul_mem c hx
-  add_mem' _ hx _ hy := S.add_mem hx hy
+def toConvexCone (C : Submodule R M) : ConvexCone R M where
+  carrier := C
+  smul_mem' c _ _ hx := C.smul_mem c hx
+  add_mem' _ hx _ hy := C.add_mem hx hy
 
-@[simp]
-theorem coe_toConvexCone (S : Submodule R M) : ↑S.toConvexCone = (S : Set M) :=
-  rfl
-
-@[simp]
-theorem mem_toConvexCone {x : M} {S : Submodule R M} : x ∈ S.toConvexCone ↔ x ∈ S :=
-  Iff.rfl
+@[simp] lemma coe_toConvexCone (C : Submodule R M) : C.toConvexCone = (C : Set M) := rfl
 
-@[simp]
-theorem toConvexCone_le_iff {S T : Submodule R M} : S.toConvexCone ≤ T.toConvexCone ↔ S ≤ T :=
-  Iff.rfl
+@[simp] lemma mem_toConvexCone : x ∈ C.toConvexCone ↔ x ∈ C := .rfl
 
 @[simp]
-theorem toConvexCone_bot : (⊥ : Submodule R M).toConvexCone = 0 :=
-  rfl
+lemma toConvexCone_le_toConvexCone : C₁.toConvexCone ≤ C₂.toConvexCone ↔ C₁ ≤ C₂ := .rfl
 
-@[simp]
-theorem toConvexCone_top : (⊤ : Submodule R M).toConvexCone = ⊤ :=
-  rfl
+@[simp] lemma toConvexCone_bot : (⊥ : Submodule R M).toConvexCone = 0 := rfl
+@[simp] lemma toConvexCone_top : (⊤ : Submodule R M).toConvexCone = ⊤ := rfl
 
 @[simp]
-theorem toConvexCone_inf (S T : Submodule R M) :
-    (S ⊓ T).toConvexCone = S.toConvexCone ⊓ T.toConvexCone :=
-  rfl
+lemma toConvexCone_inf (C₁ C₂ : Submodule R M) :
+    (C₁ ⊓ C₂).toConvexCone = C₁.toConvexCone ⊓ C₂.toConvexCone := rfl
 
 @[simp]
-theorem pointed_toConvexCone (S : Submodule R M) : S.toConvexCone.Pointed :=
-  S.zero_mem
+lemma pointed_toConvexCone (C : Submodule R M) : C.toConvexCone.Pointed := C.zero_mem
 
 end AddCommMonoid
 
 end OrderedSemiring
 
 end Submodule
 
-namespace ConvexCone
-
 /-! ### Positive cone of an ordered module -/
 
+namespace ConvexCone
 
 section PositiveCone
+variable [Semiring R] [PartialOrder R] [IsOrderedRing R] [AddCommMonoid M] [PartialOrder M]
+  [IsOrderedAddMonoid M] [Module R M] [PosSMulMono R M] {x : M}
 
-variable (R M) [Semiring R] [PartialOrder R] [IsOrderedRing R]
-  [AddCommGroup M] [PartialOrder M] [IsOrderedAddMonoid M] [Module R M] [OrderedSMul R M]
-
+variable (R M) in
 /-- The positive cone is the convex cone formed by the set of nonnegative elements in an ordered
-module.
--/
+module. -/
 def positive : ConvexCone R M where
   carrier := Set.Ici 0
   smul_mem' _ hc _ (hx : _ ≤ _) := smul_nonneg hc.le hx
   add_mem' _ (hx : _ ≤ _) _ (hy : _ ≤ _) := add_nonneg hx hy
 
-@[simp]
-theorem mem_positive {x : M} : x ∈ positive R M ↔ 0 ≤ x :=
-  Iff.rfl
+@[simp] lemma mem_positive : x ∈ positive R M ↔ 0 ≤ x := .rfl
 
+variable (R M) in
 @[simp]
 theorem coe_positive : ↑(positive R M) = Set.Ici (0 : M) :=
   rfl
 
 /-- The positive cone of an ordered module is always salient. -/
-theorem salient_positive : Salient (positive R M) := fun x xs hx hx' =>
+lemma salient_positive {M : Type*} [AddCommGroup M] [PartialOrder M] [IsOrderedAddMonoid M]
+    [Module R M] [PosSMulMono R M] : Salient (positive R M) := fun x xs hx hx' =>
```

Comment:
why do you need a new `M` here?

Reply:
because it's a group. I've changed to `G`